### PR TITLE
User can buy subscription

### PR DESCRIPTION
**Pull request by FlisAnn**
[User can buy a subscription - API](https://www.pivotaltracker.com/story/show/176031725)
### User Story
```
As a user
In order to get access to the best articles
I would like to be able to buy a subscription
```
## Changes proposed in this pull request:
* Adding stripe as a payment method
* Adding payment plan
* Adding subscriber role

## What I have learned working on this feature:
* Working with stripe
* Setting up a payment plan

## Packages/Gems added
* stripe-rails
* stripe-ruby-mock

### CODE REVIEW

File: config/stripe/plans.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This file contains descriptions of all your stripe plans
+
+# Example
+# Stripe::Plans::PRIMO #=> 'primo'
+
+Stripe.plan :DreamTimePlan do |plan|
+  # plan name as it will appear on credit card statements
```
Comment (sealfury) [inline config/stripe/plans.rb]: Maybe don't need the out-of-the-box description comments here now that the `plan` method is in place?

Comment (FlisAnn) [reply config/stripe/plans.rb]: That is a very valid point, but we believe that there is still important information in the comments that could help us when changing e.g rates.

Comment (sealfury) [reply config/stripe/plans.rb]: Fair enough

File: app/controllers/api/subscriptions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  def create
+      customer_id = get_customer(params[:stripeToken])
+      subscription = Stripe::Subscription.create({customer: customer_id, plan: 'DreamTimePlan'})
+    
+      test_env?(customer_id, subscription)
+      status = Stripe::Invoice.retrieve(subscription.latest_invoice).paid
+
+    if status
+      current_user.update_attribute(:role, 'subscriber')
+      render json: { message: 'You are now a subscriber'}, status: 201
```
Comment (emtalen) [inline app/controllers/api/subscriptions_controller.rb:13]: Please don't use the same uncreative messages that I used in the demo 🙃

File: config/stripe/plans.rb (hunk 1 above)
Comment (emtalen) [reply config/stripe/plans.rb]: I agree with William, you should clean this file from the generated comments. If you need to find something out from them- we have version control and you can always go back and look!

**Review by emtalen — CHANGES_REQUESTED** [review]

File: app/controllers/api/subscriptions_controller.rb (hunk 2 above)
Comment (FlisAnn) [reply app/controllers/api/subscriptions_controller.rb:13]: Changed it to Something went wrong. It's is very creative :)

File: config/stripe/plans.rb (hunk 1 above)
Comment (FlisAnn) [reply config/stripe/plans.rb]: Okidok, the comments have been removed

File: app/controllers/api/subscriptions_controller.rb (hunk 2 above)
Comment (emtalen) [reply app/controllers/api/subscriptions_controller.rb:13]: I meant the success message.

**Review by emtalen — APPROVED** [review]

**Pull request merged**